src/mbgl/renderer/layers/render_custom_layer.cpp

```diff
@@ -12,17 +12,16 @@ namespace mbgl {
 using namespace style;
 
 RenderCustomLayer::RenderCustomLayer(Immutable<style::CustomLayer::Impl> _impl)
-    : RenderLayer(LayerType::Custom, _impl) {
+    : RenderLayer(LayerType::Custom, _impl), host(_impl->host) {
+    host->initialize();
```

Comment:
Ahh, this is a bug. I was trying to fix a race condition with the `MGLOpenGLStyleLayer` being released when it's added to a style, and then the style is replaced before the next render frame. This is now handled by the `weakPtr` in `MGLOpenGLLayerHost`

Reply:
I moved this to the constructor in an attempt to `initialize` and retain the Obj-C peer layer sooner. This was to address a race condition where a custom layer is added to a style, but before the next frame renders, the style is swapped. In this case, the peer layer was released too soon. 
This fix didn't address that issue, but I decided to keep it in, since the render layers are created in `Renderer::Impl` and rendered in the same render pass. Adding the `assert` for protection.